### PR TITLE
More clean up.

### PR DESCRIPTION
**Pull request by agorina**

### CODE REVIEW

File: exam-processor/src/main/java/org/opentestsystem/rdw/ingest/processor/service/TDSReportProcessor.java [hunk 1]
```diff
@@ -10,8 +10,9 @@
     /**
      * Process the given {@link TDSReport}
      *
-     * @param report the report to process
+     * @param report       the report to process
      * @param tenancyChain user tenancy chain associated with import request
+     * @param importId     the id of the import record for this report
      */
-    void process(TDSReport report, final TenancyChain tenancyChain);
+    void process(TDSReport report, final TenancyChain tenancyChain, long importId);
```
Comment (agorina) [inline exam-processor/src/main/java/org/opentestsystem/rdw/ingest/processor/service/TDSReportProcessor.java]: Will remove _final_.

File: exam-processor/src/main/java/org/opentestsystem/rdw/ingest/processor/ExamProcessorConfiguration.java [hunk 2]
```diff
@@ -35,10 +35,7 @@ public void process(final Message<?> message) throws UnsupportedEncodingExceptio
         logger.info("received " + accessor.getContent() + " from " + accessor.getUserLogin());
         logger.info(accessor.getContentType() + ": " + (payload.length() > 80 ? payload.substring(0, 80) + "..." : payload));
 
-        // TODO - not sure how error processing will happen but here is the id of the import record to update
-        final Long importId = accessor.getImportId();
-        final TenancyChain tenancyChain = TenancyChain.fromString(accessor.getUserTenancyChain());
+        reportProcessor.process(tdsReportFromXml(new ByteArrayInputStream(payload.getBytes("UTF-8"))), TenancyChain.fromString(accessor.getUserTenancyChain()), accessor.getImportId());
```
Comment (malaffoon) [inline exam-processor/src/main/java/org/opentestsystem/rdw/ingest/processor/ExamProcessorConfiguration.java]: If you do the exception-based approach you wouldn't have to pass the importId in.

File: exam-processor/src/main/java/org/opentestsystem/rdw/ingest/processor/utils/ParseUtil.java [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+
+import static com.google.common.base.Strings.isNullOrEmpty;
+
+public abstract class ParseUtil {
+
+    //TODO: add comments and unit tests
```
Comment (malaffoon) [inline exam-processor/src/main/java/org/opentestsystem/rdw/ingest/processor/utils/ParseUtil.java]: uh huh

Comment (agorina) [inline exam-processor/src/main/java/org/opentestsystem/rdw/ingest/processor/utils/ParseUtil.java]: I know, life is not perfect and I am not perfect :)

File: exam-processor/src/main/java/org/opentestsystem/rdw/ingest/processor/ExamProcessorConfiguration.java (hunk 2 above)
Comment (agorina) [inline exam-processor/src/main/java/org/opentestsystem/rdw/ingest/processor/ExamProcessorConfiguration.java]: But then I need a custom exception for:  unauthorized, no assessment package and failed to parse.

Comment (malaffoon) [inline exam-processor/src/main/java/org/opentestsystem/rdw/ingest/processor/ExamProcessorConfiguration.java]: No, you'd have one exception ProcessingException (or whatever) that takes two arguments in the constructor
```
  throw new ProcessingException(ImportStatus.BAD_DATA, "invalid integer data");
```

Comment (agorina) [inline exam-processor/src/main/java/org/opentestsystem/rdw/ingest/processor/ExamProcessorConfiguration.java]: Here is what I have so far:
```
 public void process(final TDSReport report, final TenancyChain tenancyChain, long importId) {
        ImportStatus status = BAD_DATA;
        String msg = "";
        try {
            final ParsedExaminee parsedExaminee = examineeParser.parse(report.getExaminee());

            status = UNAUTHORIZED;
            authorize(tenancyChain, parsedExaminee.getSchool());

            status = UNKNOWN_ASMT;
            final Assessment assessment = assessmentService.findOneByNaturalId(report.getTest());

            status = BAD_DATA;
            studentExamProcessor.process(report.getTest(), report.getOpportunity(), studentProcessor.process(parsedExaminee), assessment);

            status = PROCESSED;

        } catch (final ParseException e) {
            msg = e.getMessage();
        } catch (Exception e) {
            msg = e.getMessage();
        }

        importRepository.updateStatus(importId, status, msg);
    }
```

Let me know if you still strongly disagree.

Comment (malaffoon) [inline exam-processor/src/main/java/org/opentestsystem/rdw/ingest/processor/ExamProcessorConfiguration.java]: Ok. Perhaps ImportException would be better than ProcessingException.

**Pull request merged**